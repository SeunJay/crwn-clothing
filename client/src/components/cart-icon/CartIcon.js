import React from "react";
import { connect } from "react-redux";
import { toggleCartHidden } from "../../redux/cart/cart.actions";
import { selectCartItemsCount } from "../../redux/cart/cart.selectors";
import { createStructuredSelector } from "reselect";
// import "./cart-icon.scss";
import {CartIconWrapper, ShoppingIcon, ItemCount} from "./cart-icon.styles"


export const CartIcon = ({ toggleCartHidden, itemCount }) => {
  return (
    <CartIconWrapper onClick={toggleCartHidden}>
      <ShoppingIcon
        className="shopping-icon"
        style={{ width: "24px", height: "24px" }}
      />
      <ItemCount className="item-count">{itemCount}</ItemCount>
    </CartIconWrapper>
  );
};

const mapStateToProps = createStructuredSelector({
  itemCount: selectCartItemsCount,
});

const mapDispatchToProps = (dispatch) => ({
  toggleCartHidden: () => dispatch(toggleCartHidden()),
});

export default connect(mapStateToProps, mapDispatchToProps)(CartIcon);
