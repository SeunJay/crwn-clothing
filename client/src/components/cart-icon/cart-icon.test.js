import React from "react";
import { shallow } from "enzyme";
import { CartIcon } from "./CartIcon";

describe("Cart icon component", () => {
  let wrapper;
  let mockToggleCartHidden;
  beforeEach(() => {
    mockToggleCartHidden = jest.fn();

    const mockProps = {
      itemCount: 5,
      toggleCartHidden: mockToggleCartHidden,
    };

    wrapper = shallow(<CartIcon {...mockProps} />);
  });

  it("should render the cart icon properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call toggleCartHidden method when icon is clicked", () => {
    const cartIcon = wrapper.find("CartIconWrapper");
    cartIcon.simulate("click");
    expect(mockToggleCartHidden).toHaveBeenCalled();
  });

  it("should render the items count as a text", () => {
    const itemCount = parseInt(wrapper.find("ItemCount").text());
    expect(itemCount).toBe(5);
  });
});


