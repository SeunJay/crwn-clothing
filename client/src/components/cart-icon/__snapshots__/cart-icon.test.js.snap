// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart icon component should render the cart icon properly 1`] = `ShallowWrapper {}`;
